modules/custom-status/custom-status.php

```diff
@@ -540,7 +540,12 @@ public function update_custom_status( $status_id, $args = [] ) {
 	 */
 	public function delete_custom_status( $status_id, $args = [] ) {
 		// Get slug for the old status
-		$old_status_slug = $this->get_custom_status_by( 'id', $status_id )->slug;
+		$old_status = $this->get_custom_status_by( 'id', $status_id );
+		if ( ! $old_status || is_wp_error( $old_status ) ) {
```

Comment:
Minor: Looking at `get_custom_status_by( ... )`, it has `WP_Error` as a return in PHPDoc but doesn't appear to actually have any cases to return an error, just `false` on failure.

Reply:
Huh, you are right. I missed that while trying to re-work those methods. It's weird that it's giving back a false on failure or an actual value on true. Lemme see where all it's used to make sure it's correctly checked and I'll update the docs too